malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java

```diff
@@ -65,12 +69,16 @@ public void addClickListener(final int position, Button infoButton) {
         infoButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                AlertDialog.Builder dialogAlert  = new AlertDialog.Builder(context);
-                dialogAlert.setMessage(drugDescriptions[position]);
-                dialogAlert.setTitle(dname[position]);
-                dialogAlert.setPositiveButton("OK", null);
-                dialogAlert.setCancelable(true);
-                dialogAlert.create().show();
+                final SpannableString s = new SpannableString(drugDescriptions[position]);
+                Linkify.addLinks(s, Linkify.ALL);
+                final AlertDialog d = new AlertDialog.Builder(getContext())
```

Comment:
Specify the reason for declaring the variables to be final or declare them as normal variables if not needed.